test/note_test.rb

```diff
@@ -2,19 +2,52 @@
 
 class TestNote < MiniTest::Test
   def setup
-    @frequency   = 220.to_f
-    @duration    = 2.0
-    @sample_rate = 22050
-    @note = MusicTheory::Note.new(frequency: @frequency, duration: @duration)
+    @test_samples = eval(File.read('test/fixtures/note_samples_array'))
+    @basic_note = MusicTheory::Note.new
+    @note_220 = MusicTheory::Note.new(frequency: 220)
+    @distorted_note = MusicTheory::Note.new(distort: true)
   end
 
-  def test_total_frames
-    frames = (@duration * @sample_rate).to_i
-    assert_equal frames, @note.total_frames
+  def test_samples_are_identical
+    assert sample_arrays_diff(@test_samples, @basic_note.samples) < (0.1 * (10 ** -10))
```

Comment:
The value of this calculation: `(0.1 * (10 ** -10)` should be stored in a constant that labels what it is. Is also slightly confusing because you're 1 decimal place away from just using scientific notation, which Ruby supports:
```ruby
original = 0.1 * (10 ** -10)
closer = 1.0 * (10 ** -11)
damn_ruby_you_convenient = 1e-11
# this is true if your ignore floating point rounding error
original == closer == damn_ruby_you_convenient
```
Another nice thing about Ruby numbers is that you can separate digits with underscores for easier reading:
```ruby
10000000 * 123
10_000_000 * 123
```